### PR TITLE
Added router component and pages

### PR DESCRIPTION
**Pull request by NZ369**
Added in pages and router component.  Since the navbar isn't setup yet, to go to a page use http://localhost:3000/Schedule

### CODE REVIEW

**Comment by Ahnaf-Ahmed** [comment]
There's a number of prettier changes suggested. A good number of them seem like theyre not neccessary but how should we deal with linting suggestions in general?

**Comment by bmazerolle** [comment]
> There's a number of prettier changes suggested. A good number of them seem like theyre not neccessary but how should we deal with linting suggestions in general?

Good point. We should add auto-prettier to the github actions and add prettier to our local pre-commit steps. I can set that up

File: app/src/UIComponents/Header.css [hunk 1]
```diff
@@ -0,0 +1,73 @@
+.header {
```
Comment (bmazerolle) [inline app/src/UIComponents/Header.css:1]: Shouldn't need any of this styling (for now) until we customize our MUI styling to match the UVic colors.

File: app/src/UIComponents/Header.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import React from 'react'
+import './Header.css';
+import { Link } from "react-router-dom";
+
+function Header() {
+
+    return (
```
Comment (bmazerolle) [inline app/src/UIComponents/Header.tsx:7]: All of this can be replaced with the Material UI navbar components/buttons, so we can delete all of this. The nav bar should also be a separate PR since you have it split off into it's own issue (https://github.com/SENG499-Company-4/Frontend/issues/29). Imo you can just leave the header out or blank for now, and implement in its own issue.

**Review by bmazerolle — CHANGES_REQUESTED** [review]
Hey @NZ369 some overall feedback:

- The routes/page formats look solid, nice job
- I don't think you are following the Atomic Design structure as outlined in the README. It should be a quick change, @sam-warren is more knowledgeable than me on this but it should just involve moving a few folders around/renaming. Use the README link as a starting point, and this [repo](https://github.com/fabiosjc/atomic-design-demo/tree/master/src) is a good reference of a demo project that follows the structure.
- As mentioned in my individual comments, I don't think you need the header at all in this PR since it's non-functional, but if we do end up keeping it, it should be re-written to use the Material UI library we imported since that will be our UI library moving forward. WE don't need to use any plain html to do UI stuff, Material UI handles all that for us.
- Please format your code using prettier by running `npm run format:fix` before your next push so the code is styled consistently. I'll add a github action to do this in the future but for now we unfortunately need to run it manually.

This is a big jump-start for our repo so thanks for writing all this code. It'll be good to go from my perspective with the above changes. If you have questions or disagree on anything just reply to any of my comments.

File: app/src/App.tsx [hunk 3]
```diff
@@ -1,22 +1,71 @@
...
-        {/* This card can go away when we have a router */}
-        <Typography variant="h4">Hello from a MUI card!</Typography>
-        <Button color="primary" variant="contained">
-          Primary Color Button
-        </Button>
-        <Button color="secondary" variant="contained">
-          Secondary Color Button
-        </Button>
-      </Card>
+      <Router>
```
Comment (sam-warren) [inline app/src/App.tsx:19]: Let's put this in a separate component called Router.tsx so our App.tsx stays lean. Root directory of app/ should be good.

File: app/src/UIComponents/Header.tsx (hunk 2 above)
Comment (sam-warren) [reply app/src/UIComponents/Header.tsx:7]: ^ https://mui.com/material-ui/react-app-bar/

File: app/src/UIComponents/Header.css (hunk 1 above)
Comment (sam-warren) [reply app/src/UIComponents/Header.css:1]: Agreed - let's try to avoid making any .css files if at all possible. I'll look into getting us a global Sass file too.

**Review by sam-warren — CHANGES_REQUESTED** [review]
Good work, just a few changes!

**Comment by NZ369** [comment]
Implemented above suggestions.  Let me know if there is anything else, thanks!

File: app/src/App.tsx [hunk 4]
```diff
@@ -1,24 +1,14 @@
...
-      </Card>
-    </ThemeProvider>
-  );
-}
-
-export default App;
+import { ThemeProvider } from '@emotion/react';
+import appTheme from 'themes/appTheme';
+import React from 'react';
+import AppRouter from './AppRouter';
```
Comment (sam-warren) [inline app/src/App.tsx:4]: Let's use absolute imports for these, instead of './AppRouter' just use 'AppRouter'

File: app/src/AppRouter.tsx [hunk 5]
```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import { BrowserRouter as Router, Routes, Route } from 'react-router-dom';
+import Header from './components/organisms/Header';
```
Comment (sam-warren) [inline app/src/AppRouter.tsx:3]: For all the imports in this file, remove the ./ at the beginning for absolute imports

**Review by sam-warren — CHANGES_REQUESTED** [review]
Much better, just one small nit. Whenever we have a local import (i.e. Import pages into the router component), let's use absolute imports supported by react instead of ./../../etc. Instead of importing from './components/organisms/somecomponent', we can drop the leading ./ to make it simply components/organisms/somecomponent.

**Comment by sam-warren** [comment]
Styled navbar, couple small changes.

Screen Shot 2022-06-03 at 2 12 22 PM

**Review by Hexuan-Z — APPROVED** [review]

**Review by Ahnaf-Ahmed — APPROVED** [review]
There was a pretty big rebase added in the last few commits. I think we should add some more explanations when doing rebases.

**Review by sam-warren — APPROVED** [review]
